### PR TITLE
Odyssey Stats: Add New badge to settings toggle

### LINKED ISSUE #28631: Odyssey Stats:  Add a "New" pillbox next to the "Enable a new Jetpack Stats experience" toggle
Image

### PR DESCRIPTION
**Pull request by a8ck3n**
Fixes #28631.

## Proposed changes:

* Adds a little "New" badge alongside the settings toggle for Odyssey Stats.
* Updates the copy to match the design.
* Moves toggle to top of list.
* Adds some margin below the Odyssey toggle.

**Before:**
StatsToggle01

**After:**
SCR-20230202-jwi

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

None

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Check out this branch.
* Prepare environment per these instructions: PejTkB-3E-p2
* Confirm the toggle has the new badge visible.

Let me know if we need to tweak the button styling or the bottom margin!

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/traffic/style.scss [hunk 1]
```diff
@@ -200,3 +200,13 @@
 		margin-top: 0.75rem;
 	}
 }
+
+.jp-stats-odyssey-badge {
+	color: $white;
+	background-color: $green-primary;
```
Comment (jsnmoon) [inline projects/plugins/jetpack/_inc/client/traffic/style.scss]: image

Overall, styling looks good! However, we probably want to stick to the design specifications and use `$blue-medium` (#3582c4) as the background color.

cc @jeffgolenski

File: projects/plugins/jetpack/_inc/client/traffic/site-stats.jsx [hunk 2]
```diff
@@ -256,8 +256,9 @@ class SiteStatsComponent extends React.Component {
 								>
```
Comment (jsnmoon) [inline projects/plugins/jetpack/_inc/client/traffic/site-stats.jsx:241]: This suggestion slightly bloats this task's scope, but could you move the Odyssey toggle above the `48-hour traffic snapshot` toggle?

File: projects/plugins/jetpack/_inc/client/traffic/style.scss (hunk 1 above)
Comment (a8ck3n) [reply projects/plugins/jetpack/_inc/client/traffic/style.scss]: Sure thing. I'll make this change today and we can update the colour again later if needed.

**Review by jsnmoon — APPROVED** [review]
Looking good, thank you! 🚢

**Comment by jeffgolenski** [comment]
Thanks y'all!

**Pull request merged**